keps/sig-architecture/3138-increase-reliability-bar/README.md

```diff
@@ -0,0 +1,415 @@
+work the amount of necessary work to get there can only increase - it will
+not decrease on its own.
+
+To make the effort trackable and sustainable we propose the following:
+1. We will limit the effort only to the test jobs that are part of
+   “master release blocking” and “master release informing” tabs.
+   - If really needed we may consider extending the set of jobs in scope,
+     but this is not a plan of record as of now.
+
+1. We will use [Sippy][] as our source of truth for determining the health
+   of the individual jobs and tests.
+   We’ve already set up [Kubernetes Sippy][] that is monitoring the jobs
+   described above.
+   We will focus on the “Top failing tests” part of it.
+
+1. People may use https://storage.googleapis.com/k8s-triage/index.html for
+   finding instances of the failing tests that they can use for debugging purposes.
+
+1. We will be periodically looking at the “Top failing parts” to:
+   - Open issues (labeled with “kind/reliability-issue” and appropriate SIG)
+     for the worst one such that:
+     - There should be at most 3 open issues for a given SIG, with the caveat
+       of significant regressions - if the test pass rate goes down by more
+       than 10% week over week, that justifies opening additional ones
+     - We considered automatic issue filing but decided not to start with it due to:
+       - CI signal experience with flakes (e.g. system-level issues, handling
+         duplicates etc.)
+       - people tending to ignore bot-originating issues
+     - The proper labeling of issues would mean that people would be able to easily
+       find issues currently opened for a given SIG, via a simple github query like
+       this `https://github.com/kubernetes/kubernetes/issues?q=is%3Aopen+is%3Aissue+label%3Asig%2Ffoo+label%3Akind%2Freliability-issue`
+   - Notify the SIG on Slack asking for explicit acknowledgement on the issue
+   - Start a timer on the issue: 2 months (half of the release) is the proposed default
+
+1. Even though each test is generally assigned to a SIG, tests may be failing
+   due to bugs or changes done by other SIGs. SIG labels are designed to be
+   easy to apply and change to ensure a collaborative nature of issue triage.
+   If a SIG will show that the bug is elsewhere, they should notify that
+   other SIG, reassign the bug and the new owning SIG is allowed to restart
+   the clock.
+
+1. The issues can’t simply be closed with “test is now passing”. Even if in
+   the meantime the failure rate dropped, it should still be investigated
+   and debugged to ensure that this won’t spike again in the future for the
+   same reason.
+   If the test can’t be debugged due to lack of data, the expected step
+   forward is to increase our observability by adding logs, metrics, events,
+   etc. to allow debugging the future instances of the flake.
+
+1. Once debugged and fixed, the issue should be labeled with the root cause
```

Comment:
IME of fixing Kubelet e2es lately, it would be useful to also explicitly track issues where features were modified but tests were not updated properly.

When optional jobs are flaky, a lot of legitimate failures get missed because folks stop reading the results.

Reply:
>  a lot of legitimate failures get missed because folks stop reading the results.

what folks do you refer here? I think that each SIG is responsible of reading their results, there is some sig-release group that monitor the critical jobs, but not the optional ones

Reply:
> what folks do you refer here? I think that each SIG is responsible of reading their results,

We have so many jobs now, that there are many that noone is really looking at (or looking to the extent to find flakes - rather checking if something isn't hard down).

Reply:
yeah, job ownership is another problem